plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -303,9 +306,26 @@ defineBuiltinTerms = do
     defineBuiltinTerm 'Builtins.appendString $ mkBuiltin PLC.AppendString
     defineBuiltinTerm 'Builtins.emptyString $ PIR.mkConstant () ("" :: Text)
     defineBuiltinTerm 'Builtins.equalsString $ mkBuiltin PLC.EqualsString
-    defineBuiltinTerm 'Builtins.trace $ mkBuiltin PLC.Trace
     defineBuiltinTerm 'Builtins.encodeUtf8 $ mkBuiltin PLC.EncodeUtf8
 
+    -- Tracing
+    defineBuiltinTerm 'Builtins.trace $
+        -- When `remove-trace` is specified, we define `trace` as `\_ a -> a` instead of the builtin version.
+        -- TODO: when simpilers are enabled, we should reduce and inline a `\_ a -> a` call to just `a`.
+        -- Reducing `test/Plugin/NoTrace/traceComplex.plc.golden` is a good start.
```

Comment:
Not a TODO that makes sense in _this_ location!

Reply:
Sorry, I've removed and rebased. My memory is so sad I even write code todos on refrigerator sticky notes nowadays...